base/bucket.go

```diff
@@ -606,17 +607,19 @@ func IsKeyNotFoundError(bucket Bucket, err error) bool {
 		return false
 	}
 
-	if err == gocb.ErrKeyNotFound {
+	unwrappedErr := pkgerrors.Cause(err)
```

Comment:
@adamcfraser this could also be changed to:

err = pkgerrors.Cause(err)

Eg, shadow the variable with a different type, which I think might be more idiomatic Go.  Let me know what you think.